### ISSUE TITLE
Zoom font with window improvement - an off switch (patch)

### ISSUE THREAD

**Issue by morkk** [mintty/mintty#476]
I use Autohotkey macros to move and resize windows and happen to have picked Alt-Shift-? as the shortcut keys; thus the font always zooms for me and that isn't what I want.  So here's a patch to switch the zoom_font_with_window behaviour off in the hope that I won't have to keep applying it to new versions.  And, yes, I did first try to adapt my ahk script but to no avail and the shortcuts are now muscle memory.

cheers
mark
```
--- config.c.old        2015-09-02 00:04:17.213447700 +1000
+++ config.c    2015-09-02 00:04:22.788766600 +1000
@@ -30,6 +30,7 @@
   // Text
   .font = {.name = "Lucida Console", .isbold = false, .size = 9},
   .font_smoothing = FS_DEFAULT,
+  .zoom_font_with_window = true,
   .bold_as_font = -1,  // -1 means "the opposite of bold_as_colour"
   .bold_as_colour = true,
   .allow_blinking = false,
@@ -150,6 +151,7 @@
   {"AllowBlinking", OPT_BOOL, offcfg(allow_blinking)},
   {"Locale", OPT_STRING, offcfg(locale)},
   {"Charset", OPT_STRING, offcfg(charset)},
+  {"ZoomFontWithWindow", OPT_BOOL, offcfg(zoom_font_with_window)},

   // Keys
   {"BackspaceSendsBS", OPT_BOOL, offcfg(backspace_sends_bs)},
--- config.h.old        2015-09-02 00:03:54.473147100 +1000
+++ config.h    2015-09-02 00:03:59.004406200 +1000
@@ -50,6 +50,7 @@
   // Text
   font_spec font;
   char font_smoothing;
+  bool zoom_font_with_window;
   char bold_as_font;    // 0 = false, 1 = true, -1 = undefined
   bool bold_as_colour;
   bool allow_blinking;
--- winmain.c.old       2015-09-02 00:03:29.509719200 +1000
+++ winmain.c   2015-09-02 00:03:34.568008600 +1000
@@ -855,7 +855,7 @@
         if (!(GetKeyState(VK_LWIN) & 0x80))
           zoom_token = 1;
 #endif
-        bool scale_font = (zoom_token > 0) && (GetKeyState(VK_SHIFT) & 0x80);
+        bool scale_font = cfg.zoom_font_with_window && (zoom_token > 0) && (GetKeyState(VK_SHIFT) & 0x80);
         win_adapt_term_size(false, scale_font);
         if (zoom_token > 0)
           zoom_token = zoom_token >> 1;
```

**Comment by mintty** [comment]
(Fixed your comment layout by embedding code in ```...```)
This would affect all “zoom font with window” functions, including not only Shift-mouse-drag/-max-/-minimize (which can probably not be distinguished from your hotkey) but also Shift-Alt-Enter/-F11.
Maybe I'll try to craft a patch that recognizes the latter to keep them enabled; or maybe you can try to find a way?

**Comment by mintty** [comment]
Patch uploaded. Please test.

**Comment by morkk** [comment]
thanks for adding my patch; sorry for missing the docco - seems to happen to me a lot :-)

it's hard to get around my particular situation because the shift key is definitely down when the resize message is received, so this switch was my only solution

**Comment by mintty** [comment]
Released in 2.2.1.
